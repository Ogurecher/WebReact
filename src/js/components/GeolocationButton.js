import React, { Component } from "react";
import { connect } from "react-redux";
import { getLocation } from "../actions/index";

export class GeolocationButton extends Component {
    constructor(props) {
        super(props);

        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleSubmit(event) {
        event.preventDefault();
        this.props.getLocation();
    }

    componentDidMount() {
        this.props.getLocation();
    }

    render() {
        return (
            <div>
                <form onSubmit={this.handleSubmit}>
                    <input type='submit' value = 'Get location'/>
                </form>
                
            </div>
        );
    }
}

function mapStateToProps(state) {
    return {
        position: state.position
    };
}

export default connect(
    mapStateToProps,
    { getLocation }
  )(GeolocationButton);