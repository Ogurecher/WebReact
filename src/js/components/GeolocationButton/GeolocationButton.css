.geolocation-wrapper {
    display: flex;
    flex-direction: row;
    align-items: baseline;
}

.geolocation-text {
    font-size: 200%;
    margin-left: 1%;
    margin-right: 1%;
}