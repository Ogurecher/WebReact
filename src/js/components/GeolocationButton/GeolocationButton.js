import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getLocation } from '../../actions/index';
import './GeolocationButton.css';

export class GeolocationButton extends Component {
    constructor(props) {
        super(props);

        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleSubmit(event) {
        event.preventDefault();
        this.props.getLocation();
    }

    componentDidMount() {
        this.props.getLocation();
    }

    render() {
        return (
            <div className='geolocation-wrapper'>
                <p className='geolocation-text'>Weather here</p>
                <form onSubmit={this.handleSubmit}>
                    <input type='submit' value = 'Get location'/>
                </form>
            </div>
        );
    }
}

function mapStateToProps(state) {
    return {};
}

export default connect(
    mapStateToProps,
    { getLocation }
  )(GeolocationButton);