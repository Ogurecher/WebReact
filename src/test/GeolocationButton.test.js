import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';

import GeolocationButton from '../js/components/GeolocationButton/GeolocationButton';

const middlewares = [thunk];
const store = configureMockStore(middlewares);

describe('GeolocationButton', function() {
    test('snapshot renders', function() {
        const storeMock = store();

        const geolocationMock = {
            getCurrentPosition: jest.fn()
        };
        global.navigator.geolocation = geolocationMock;

        const component = renderer.create(<Provider store={storeMock} ><GeolocationButton /></Provider>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});