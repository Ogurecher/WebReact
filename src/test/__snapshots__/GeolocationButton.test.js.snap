// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeolocationButton snapshot renders 1`] = `
<div
  className="geolocation-wrapper"
>
  <p
    className="geolocation-text"
  >
    Weather here
  </p>
  <form
    onSubmit={[Function]}
  >
    <input
      type="submit"
      value="Get location"
    />
  </form>
</div>
`;
